Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -93,8 +236,34 @@ public class DocumentIterator: IteratorProtocol {
         return values
     }
 
+    // uses an iterator to copy (key, value) pairs of the provided document from range [startIndex, endIndex) into a new
+    // document. starts at the startIndex-th pair and ends at the end of the document or the (endIndex-1)th index, 
+    // whichever comes first. 
+    internal static func subsequence(of doc: Document, startIndex: Int = 0, endIndex: Int = Int.max) -> Document {
+        precondition(endIndex >= startIndex, "endIndex must be >= startIndex")
+
+        guard let iter = DocumentIterator(forDocument: doc) else { return [:] }
```

Comment:
not really sure what to do in this case. none of the methods calling subsequence can throw. maybe just preconditionFailure? idk

Reply:
I suppose consistency with Swift is the best we can do. I assume that means precondition failures?

Reply:
well, in general we would do that when we can't error, but this reminds me of the cases in `.extendedJSON` and `.canonicalExtendedJSON` where we opted out of a `preconditionFailure` and instead returned an empty string if parsing the data failed, because someone could call the method on unvalidated/corrupt BSON and have a crash.

similarly, here we could experience
```swift
let doc: Document = Document(fromBSON: corrupt_bson_data_here)
print(doc.suffix(maxLength: 1)) // precondition failure
```